### ISSUE TITLE
`cifar10_cnn.py` example fails to converge using `rmsprop` optimizer

### ISSUE THREAD

**Issue by JefferyRPrice** [fchollet/keras#4897]
The `cifar10_cnn.py` example as it is fails to converge. If `rmsprop` is changed to `adam` or replaced with an `SGD(lr=0.01, momentum=0.9, decay=1e-6, nesterov=True)` optimizer, the example works fine. Examples of the history for the 200 epochs in the example are shown below. Other than storing and dumping the results to file via `cPickle` to produce the plots, nothing else in `cifar10_cnn.py` was changed excepting the optimizers as stated.

Theano backend, GPU mode, Ubuntu 14.04, running on an AWS g2.2xlarge instance ("G2 Instances are backed by 1 x NVIDIA GRID GPU (Kepler GK104) and 8 x hardware hyperthreads from an Intel Xeon E5-2670").

(Also thanks for making Keras with such a great API.)

![rmsprop](https://cloud.githubusercontent.com/assets/1270864/21594364/bfca46cc-d0ef-11e6-97df-0dcc2442de00.png)
![adam](https://cloud.githubusercontent.com/assets/1270864/21594369/c7ec2780-d0ef-11e6-8ed6-b9d47b0162b4.png)
![sgd](https://cloud.githubusercontent.com/assets/1270864/21594370/ca3513d0-d0ef-11e6-94bb-a70a4d037951.png)

---

Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [X] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [X] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [X] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [X] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by surfreta** [comment]
Hi JefferyRPrice, may I know how you can save the performance metrics of train_loss, train_acc, val_loss, val_acc during the training process, and feed them into a plot. Would you like to share the script, directly related to this part? Thanks.

**Comment by JefferyRPrice** [comment]
I modified `cifar10_cnn.py` that just imports `cPickle`, adds `results = model.fit*` to both fits (with and without augmentation), and then dumps `results.history` to a file. [Here's the one](https://gist.github.com/JefferyRPrice/e29ce832cd6c4e93d19f04bfa4d50ba7) that used SGD.

To plot, I used [this code](https://gist.github.com/JefferyRPrice/6cc6a0467a66e95d192e751654ccc8c9) in an Jupyter Python 2 notebook.

For what it's worth, I did the experiment both with and without data augmentation with the same basic results, where `rmsprop` was wonky and `adam` and `SGD` were fine.

**Comment by surfreta** [comment]
Thank you so much for sharing the code for dumping the result. I was also trying to generate the plot as you show. By the way, the metrics of loss corresponds to score, or cost, or objective ( as shown here at  https://keras.io/objectives/ ). But I was not able to find the definition of acc, e.g., accuracy. I can imagine what does it mean, but I want to know how does it defined. Do you happen to have any information on this? Thanks.

**Comment by redst4r** [comment]
standard definition of classification accuracy I guess:
`ACC = TP+TN/(P+N)`

or for multiclass, just the trace of the confusion matrix C divided by number of samples:
`np.trace(C)/np.sum(C)`

**Comment by JefferyRPrice** [comment]
Changing the `rmsprop` parameters to `RMSprop(lr=0.0001, rho=0.95)` results in convergence (see attached plot, which was generated with `data_augmentation = False`).

I would propose that `rmsprop` in the example be changed to `adam` so that the example works out-of-the-box.

![cifar-rmsprop-mod](https://cloud.githubusercontent.com/assets/1270864/21630250/7b4a37f0-d1fd-11e6-89f4-cc43548845b1.png)
